### PR TITLE
Support Canary for ECS without ELB or target groups

### PR DESCRIPTION
**Pull request by t-kikuc**
**What this PR does / why we need it**:

This is the simplest way to realize canary for ECS without target groups.

This PR supports the below stages (almost the same as the ELB case):
- CanaryRollout
- PrimaryRollout
- CanaryClean
- Rollback

NOTE: The TrafficRouting stage and Blue/Green Deployment are not supported due to ECS Service Discovery's constraints.

**Which issue(s) this PR fixes**:

Part of  #4616 
( or fixes )

**Does this PR introduce a user-facing change?**: no (new feature)

- **How are users affected by this change**: no
- **Is this breaking change**: no
- **How to migrate (if breaking change)**: no

### CODE REVIEW

**Comment by t-kikuc** [comment]
@khanhtc1202 
Hi, I tested and it worked well.

File: pkg/app/piped/executor/ecs/rollback.go [hunk 1]
```diff
@@ -86,6 +86,8 @@ func (e *rollbackExecutor) ensureRollback(ctx context.Context) model.StageStatus
 	if !ok {
 		return model.StageStatus_STAGE_FAILURE
 	}
+	// Even if target groups are not found, we can continue to rollback.
+	// So the accessType does not matter here.
```
Comment (khanhtc1202) [inline pkg/app/piped/executor/ecs/rollback.go:90]: Should not mention unchanged thing here, let's not add confusing comments.

File: pkg/app/piped/executor/ecs/deploy.go [hunk 2]
```diff
@@ -152,23 +163,40 @@ func (e *deployExecutor) ensureCanaryRollout(ctx context.Context) model.StageSta
...
+	default:
+		e.LogPersister.Errorf("Unsupported access type %s in stage %s for ECS application", e.appCfg.Input.AccessType, e.Stage.Name)
 		return model.StageStatus_STAGE_FAILURE
 	}
 
 	return model.StageStatus_STAGE_SUCCESS
 }
 
 func (e *deployExecutor) ensureTrafficRouting(ctx context.Context) model.StageStatus {
+	// Traffic Routing is not supported for ECS Service Discovery.
```
Comment (khanhtc1202) [inline pkg/app/piped/executor/ecs/deploy.go:194]:
```suggestion
	// Traffic Routing is not supported for other kind than ELB.
```

File: pkg/app/piped/executor/ecs/rollback.go (hunk 1 above)
Comment (khanhtc1202) [inline pkg/app/piped/executor/ecs/rollback.go:90]:
```suggestion
```

**Comment by t-kikuc** [comment]
@khanhtc1202 
Thank you for your comments, and I fixed them.

**Review by khanhtc1202 — APPROVED** [review]
great, thank you 🫡

**Pull request merged**